man/swun.1
.TH SWUN 1 "2012 Juli 22"

.SH NAME
swun \- simple SoftWare Update Notifier

.SH SYNOPSIS
.br
.B swun
.PP

.SH DESCRIPTION
.B swun 
is a very simple update notifier intended for the use with Slackware Linux.
.PP
swun places itself as an an icon into the notification area of the 
desktop environments. It uses 
.B slackpkg
as backend, so you must have configured slackpkg!
Swun does not more than check for changes in the changelog 
with 
.B "slackpkg check-updates" 
and changes the icon color accordingly:
.IP 
.BR red 
== update available,
.IP
.BR green 
== system is up to date(no changes to the changelog).

.SS
Left Click:
.TP 5
Shows the information from the last command. 
That means you normally get the status.

.SS
Middle Click:
.TP 5
Shows Version information.

.SS
Right Click: Menu Actions
.TP 5
- Check Changelog
this is the action, that is automatically triggered from time to time, 
but you are able to trigger it here.
.TP 5
- Check Update State
this will do a "Check Changelog" and check afterwards, if all packages
listed in the cache are really installed (
.B slackpkg upgrade
, without install, nevertheless root rights are required). 
if not all upgradeable packages are installed, it will display these items.
And the icon color will be 
.BR orange.
You are able to check this special update state only with this action.
.TP 5
- Update System
This will open a terminal with root rights to give you the chance 
to run an interactive update (
.B slackpkg upgrade
). 
No matter if you install all upgrades or only some of them, the state will
be up-to-date(green) afterwards, because the changelog is up-to-date.
Detailed information can be retrieved with "Check Update State".
.TP 5
- Help 
This opens a terminal with this man page.
.TP 5
- Quit 
is self explaining ;-)

.SH NOTES
.SS
Note on root rights
You can also add slackpkg tool as a sudo command:
# enable sudo package management without passwords questions like in this line:
$(user)  ALL=(root) NOPASSWD: /usr/sbin/slackpkg
And set the "USE_SUDO" variable in swun.sh to "yes".

.SH OPTIONS
Currently there are no options.

.SH SEE ALSO
slackpkg(8)

.SH AUTHOR
.B swun
was made by Benno <fwxx@rocketmail.com>


ChangeLog
2013-02-17 11:46:01 +0100 Benno <fwxx@rocketmail.com> 

	* added help menu entry (HEAD, origin/master, master)

2013-02-17 11:43:17 +0100 Benno <fwxx@rocketmail.com> 

	* added GenericName to desktop file

2013-02-17 11:42:14 +0100 Benno <fwxx@rocketmail.com> 

	* added section building and installing to README, fixed slackware build scripts

2012-08-01 21:43:58 +0200 Benno <fwxx@rocketmail.com> 

	* fixed typo in TAG, TMP now also does not point to SBo subdirectory

2012-08-01 21:36:36 +0200 Benno <fwxx@rocketmail.com> 

	* default package tag set to fwxx, as it is still not in slackbuilds.org, changed version to 0.0.2 now

2012-08-01 21:25:14 +0200 Benno <fwxx@rocketmail.com> 

	* manually triggerd checkChangelog will always show a message as result. this gives a feedback to the user, that something happened

2012-07-28 00:00:11 +0200 Benno <fwxx@rocketmail.com> 

	* nearly completely rewritten README

2012-07-26 19:40:34 +0200 Benno <fwxx@rocketmail.com> 

	* updated slackware package info (v0.0.1)

2012-07-24 19:24:33 +0200 Benno <fwxx@rocketmail.com> 

	* fix in direct build script

2012-07-24 19:20:29 +0200 Benno <fwxx@rocketmail.com> 

	* updated man page, little behaviour change

2012-07-23 23:53:15 +0200 Benno <fwxx@rocketmail.com> 

	* fixed wrong quotes

2012-07-23 23:47:47 +0200 Benno <fwxx@rocketmail.com> 

	* added first version of a man page

2012-07-23 00:03:20 +0200 Benno <fwxx@rocketmail.com> 

	* initial checkin



swun.sh
#!/bin/sh
###############################################################################
#
# Copyright (c) 2012, Benno Grolik <fwxx@rocketmail.com>
# All rights reserved.
# 
# Redistribution and use in source and binary forms, with or without 
# modification, are permitted provided that the following conditions are met:
# 
# - Redistributions of source code must retain the above copyright notice, 
#     this list of conditions and the following disclaimer.
# - Redistributions in binary form must reproduce the above copyright notice, 
#     this list of conditions and the following disclaimer in the documentation 
#     and/or other materials provided with the distribution.
# - Neither the name of Benno Grolik nor the names of its contributors may be 
#     used to endorse or promote products derived from this software without 
#     specific prior written permission.
# 
# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE 
# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF 
# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
# THE POSSIBILITY OF SUCH DAMAGE.
#
###############################################################################

# @PARAM USE SUDO:
# set it to "yes" and put a line like the following into /etc/sudoers to 
# enable sudo package management without passwords questions
# $(user)  ALL=(root) NOPASSWD: /usr/sbin/slackpkg
# 
USE_SUDO="no"
###############################################################################
PKG_TOOL="/usr/sbin/slackpkg"
TERMINAL="/usr/bin/xterm -bg black -fg white"
LOG="/tmp/swun.log"
###############################################################################
SUDO=""
if [ "$USE_SUDO" == "yes" ]; then
    SUDO="/usr/bin/sudo"
fi
#echo "sudo=$SUDO"




function doCheck()
{
    #avoid control character with -spinning=off
    OUTPUT=$(/usr/sbin/slackpkg -spinning=off check-updates | grep -v "^$")
    echo "$OUTPUT"
    if [ "$OUTPUT" == "News on ChangeLog.txt" ]; then
	RETURN=1;
    else
	RETURN=0;
    fi
    return $RETURN;
}


function checkPackages()
{
#    OUTPUT=$($PKG_TOOL -dialog=off -batch=on -default_answer=n upgrade-all | grep -v "^$" | grep -v "DONE$" | grep -v "^Total package" | grep -v "^Do you wish to upgrade")
    OUTPUT=$($SUDO $PKG_TOOL -dialog=off -batch=on -default_answer=n upgrade-all | grep '\.t.z')
    for line in $OUTPUT 
    do
	echo $line
    done
    if [ -z "$OUTPUT" ]; then
	RESULT=0
    else
	RESULT=1
    fi
    return $RESULT;
}

function autoUpdate()
{
    $SUDO $PKG_TOOL -batch=on -default_answer=y update
    return $?
}


function autoUpgrade()
{
    $SUDO $PKG_TOOL -dialog=off -batch=on -default_answer=y upgrade-all
    return $?
}


function interactiveUpgrade()
{
    $SUDO $PKG_TOOL -batch=on -default_answer=y update      2>  "$LOG"
    $SUDO $PKG_TOOL upgrade-all 2>> "$LOG"
    RESULT=$?
    exit $RESULT;
}




###############################################################################
#main
case "$1" in
    "check")
	doCheck
	;;
    "checkPackages")
	if [ "$USE_SUDO" == "yes" ]; then
	    checkPackages
	else
	kdesu --noignorebutton -t -c "$0 root checkPackages" 2>/dev/null
        fi
	;;
    "autoUpdate")
	if [ "$USE_SUDO" == "yes" ]; then
	    autoUpdate
	else
	kdesu --noignorebutton -t -c "$0 root autoUpdate"    2>/dev/null
        fi
	;;
    "autoUpgrade")
	if [ "$USE_SUDO" == "yes" ]; then
	    autoUpgrade
	else
	kdesu --noignorebutton -t -c "$0 autoUpgrade" 2>/dev/null
        fi
	;;
    "upgrade")
	rm -f "$LOG"
	touch "$LOG"
	if [ "$USE_SUDO" == "yes" ]; then
	    $TERMINAL -e "$0 root interactiveUpgrade; sleep 3"
	else
	kdesu --noignorebutton -c "$TERMINAL -e \"$0 root interactiveUpgrade; sleep 3\"" 2>/dev/null
        fi
	RESULT=$?
	cat "$LOG"
	rm -f "$LOG"
	exit $RESULT
	;;
    "root")
	case "$2" in
	    "checkPackages")
		checkPackages
		;;
	    "autoUpdate")
		autoUpdate
		;;
	    "autoUpgrade")
		autoUpgrade
		;;
	    "interactiveUpgrade")
		interactiveUpgrade
		;;
	esac
	;;
    *)
	echo "usage: $0 [check|checkPackages|autoUpdate|upgrade|autoUpgrade]"
	;;
esac
